modules/effects/src/shader-modules/fxaa/fxaa.js

```diff
@@ -0,0 +1,681 @@
+// https://github.com/AnalyticalGraphicsInc/cesium/blob/master/LICENSE.md
+// with the following modifications:
+// - fxaa_sampleColor to integrate with luma.gl
+// - Return color value with alpha to avoid another tap
+// - Calculate luminance in FxaaLuma
+
+// COMMENTS FROM CESIUM VERSION
+// NVIDIA GameWorks Graphics Samples GitHub link: https://github.com/NVIDIAGameWorks/GraphicsSamples
+// Original FXAA 3.11 shader link: https://github.com/NVIDIAGameWorks/GraphicsSamples/blob/master/samples/es3-kepler/FXAA/FXAA3_11.h
+//
+// Steps used to integrate into Cesium:
+// * The following defines are set:
+//       #define FXAA_PC 1
+//       #define FXAA_WEBGL_1 1
+//       #define FXAA_GREEN_AS_LUMA 1
+//       #define FXAA_EARLY_EXIT 1
+//       #define FXAA_GLSL_120 1
+// * All other preprocessor directives besides the FXAA_QUALITY__P* directives were removed.
+// * Double underscores are invalid for preprocessor directives so replace them with a single underscore. Replace
+//   /FXAA_QUALITY__P(.*)/g with /FXAA_QUALITY__P$1/.
+// * There are no implicit conversions from ivec* to vec* so replace:
+//       #define FxaaInt2 ivec2
+//           with
+//       #define FxaaInt2 vec2
+// * The texture2DLod function is only available in vertex shaders so replace:
+//       #define FxaaTexTop(t, p) texture2DLod(t, p, 0.0)
+//       #define FxaaTexOff(t, p, o, r) texture2DLod(t, p + (o * r), 0.0)
+//           with
+//       #define FxaaTexTop(t, p) texture2D(t, p)
+//       #define FxaaTexOff(t, p, o, r) texture2D(t, p + (o * r))
+// * FXAA_QUALITY_PRESET is prepended in the javascript code. We may want to expose that setting in the future.
+// * The following parameters to FxaaPixelShader_ are unused and can be removed:
+//       fxaaConsolePosPos
+//       fxaaConsoleRcpFrameOpt
+//       fxaaConsoleRcpFrameOpt2
+//       fxaaConsole360RcpFrameOpt2
+//       fxaaConsoleEdgeSharpness
+//       fxaaConsoleEdgeThreshold
+//       fxaaConsoleEdgeThresholdMi
+//       fxaaConsole360ConstDir
+//
+// Choose the quality preset.
+// This needs to be compiled into the shader as it effects code.
+// Best option to include multiple presets is to
+// in each shader define the preset, then include this file.
+//
+// OPTIONS
+// -----------------------------------------------------------------------
+// 10 to 15 - default medium dither (10=fastest, 15=highest quality)
+// 20 to 29 - less dither, more expensive (20=fastest, 29=highest quality)
+// 39       - no dither, very expensive
+//
+// NOTES
+// -----------------------------------------------------------------------
+// 12 = slightly faster then FXAA 3.9 and higher edge quality (default)
+// 13 = about same speed as FXAA 3.9 and better than 12
+// 23 = closest to FXAA 3.9 visually and performance wise
+//  _ = the lowest digit is directly related to performance
+// _  = the highest digit is directly related to style
+//
+
+const fs = `
+#define FXAA_QUALITY_PRESET 29
```

Comment:
Should this be supplied by the user of this module?

Reply:
Could be, but I've found the difference to be fairly subtle. Don't feel it would really be a meaningful option.